scripts/start-setupVanillaTweaks

```diff
@@ -12,73 +12,149 @@ set -e -o pipefail
+  CRAFTINGTWEAKS=$(jq -jc '.packs // empty' $VT_FILE)
+  if [ ! "$CRAFTINGTWEAKS" ]; then
+    log "ERROR: unable to retrieve crafting tweaks from ${VT_FILE}"
+    exit 2
+  fi
+  VT_ZIPDATA_URL="https://vanillatweaks.net/assets/server/zipcraftingtweaks.php"
+  
+  DOWNLOAD_URL=$(curl -X POST -F "packs=${CRAFTINGTWEAKS}" -F "version=${VT_VERSION}" $VT_ZIPDATA_URL | jq -r '.link // empty')
   if [ ! "$DOWNLOAD_URL" ]; then
-    log "ERROR: unable to retrieve DOWNLOAD_URL from vanillatweaks.net!"
+    log "ERROR: unable to retrieve crafting tweak DOWNLOAD_URL from vanillatweaks.net!"
     exit 2
   fi
 
-  TEMPZIP=/tmp/vanillatweaks.zip
-  if ! get -o $TEMPZIP "https://vanillatweaks.net${DOWNLOAD_URL}"; then 
-    log "ERROR: failed to download from ${DOWNLOAD_URL}"
+  mkdir -p "$DATAPACKS_DIR"
+  if ! get -o $DATAPACKS_DIR/crafting_tweaks.zip "https://vanillatweaks.net${DOWNLOAD_URL}"; then
+    log "ERROR: failed to download crafting tweaks from ${DOWNLOAD_URL}"
+    exit 2
+  fi
+}
+
+# Download resource packs
+downloadResourcepacks(){
+  VT_FILE=$1
+  RESOURCEPACKS=$(jq -jc '.packs // empty' $VT_FILE)
+  if [ ! "$RESOURCEPACKS" ]; then
+    log "ERROR: unable to retrieve resourcepacks from ${VT_FILE}"
     exit 2
   fi
+  VT_ZIPDATA_URL="https://vanillatweaks.net/assets/server/zipresourcepacks.php"
 
-  mkdir -p "$out_dir"
-  if ! unzip -o -d "$out_dir" $TEMPZIP; then
-    log "ERROR: failed to unzip the ${PACKS} from ${$TEMPZIP}"
+  DOWNLOAD_URL=$(curl -X POST -F "packs=${RESOURCEPACKS}" -F "version=${VT_VERSION}" $VT_ZIPDATA_URL | jq -r '.link // empty')
+  if [ ! "$DOWNLOAD_URL" ]; then
+    log "ERROR: unable to retrieve resourcepacks DOWNLOAD_URL from vanillatweaks.net!"
+    exit 2
   fi
 
-  # clean up files time!
-  rm -f $TEMPZIP
-  # cleans up temp vanilla tweaks file download to get stored packs
-  if [[ "$VANILLATWEAKS_SHARECODE" ]]; then
-    rm -f $VANILLATWEAKS_FILE
+  mkdir -p "$RESOURCEPACKS_DIR"
+  if ! get -o $RESOURCEPACKS_DIR/resourcepacks.zip "https://vanillatweaks.net${DOWNLOAD_URL}"; then
+    log "ERROR: failed to download resourcepacks from ${DOWNLOAD_URL}"
+    exit 2
   fi
+}
+
+# Example: VANILLATWEAKS_SHARECODE=MGr52E
+# Code generated from the UI website, typically a alphanumeric 6 digit code.
+if [[ "$VANILLATWEAKS_SHARECODE" ]]; then
+  VANILLATWEAKS_FILE=()
+  for SHARECODE in ${VANILLATWEAKS_SHARECODE//,/ }; do
+    TMP_FILE="/tmp/${SHARECODE}.json"
+    SHARECODE_LOOKUP_URL="https://vanillatweaks.net/assets/server/sharecode.php?code=${SHARECODE}"
+    curl -f $SHARECODE_LOOKUP_URL -o $TMP_FILE
+    if [ ! -f "$TMP_FILE" ]; then
+      log "ERROR: Unable to use ${SHARECODE} share code provided to retreive vanillatweaks file"
+      exit 2
+    fi
+    VANILLATWEAKS_FILE+="${TMP_FILE},"
+  done
+  log "VANILLATWEAKS_FILE: ${VANILLATWEAKS_FILE}"
+fi
+
+# Use vanillatweaks file to specify VT and datapacks and crafting tweaks
+if [[ "$VANILLATWEAKS_FILE" ]]; then
+  for VT_FILE in ${VANILLATWEAKS_FILE//,/ }; do
```

Comment:
When using `VANILLATWEAKS_SHARECODE`, it looks like `VANILLATWEAKS_FILE` gets set as an array. I'm surprised bash allows the text replacement on an array, but the tests pass.